backends/ebpf/ebpfType.cpp

```diff
@@ -384,4 +389,59 @@ void EBPFMethodDeclaration::emit(CodeBuilder *builder) {
     builder->newline();
 }
 
+unsigned EBPFScalarTypePNA::alignment() const {
+    if (width <= 8)
+        return 1;
+    else if (width <= 16)
+        return 2;
+    else if (width <= 24)
+        return 1;  // compiled as u8*
+    else if (width <= 32)
+        return 4;
+    else if (width <= 56)
+        return 1;  // compiled as u8*
+    else if (width <= 64)
+        return 8;
+    else
+        // compiled as u8*
+        return 1;
+}
+
+void EBPFScalarTypePNA::declare(CodeBuilder *builder, cstring id, bool asPointer) {
+    if (generatesScalar(width) && isPrimitiveByteAligned == true) {
+        emit(builder);
+        if (asPointer) builder->append("*");
+        builder->spc();
+        builder->append(id);
+    } else {
+        if (asPointer)
+            builder->appendFormat("u8* %s", id.c_str());
+        else
+            builder->appendFormat("u8 %s[%d]", id.c_str(), bytesRequired());
+    }
+}
+
+void EBPFScalarTypePNA::declareInit(CodeBuilder *builder, cstring id, bool asPointer) {
+    if (generatesScalar(width) && isPrimitiveByteAligned == true) {
+        emit(builder);
+        if (asPointer) builder->append("*");
+        builder->spc();
+        id = id + cstring(" = 0");
+        builder->append(id);
+    } else {
+        if (asPointer)
+            builder->appendFormat("u8* %s = NULL", id.c_str());
+        else
+            builder->appendFormat("u8 %s[%d] = {0}", id.c_str(), bytesRequired());
+    }
+}
+
+void EBPFScalarTypePNA::emitInitializer(CodeBuilder *builder) {
+    if (generatesScalar(width) && isPrimitiveByteAligned == true) {
+        builder->append("0");
+    } else {
+        builder->append("{ 0 }");
```

Comment:
may be you can add a comment that array initializer is emitted for non byte aligned primitive types as they are emitted as array type.